src/Compilers/CSharp/Portable/Binder/Binder_Constraints.cs

```diff
@@ -210,6 +204,21 @@ internal partial class Binder
                                                 continue;
                                         }
                                     }
+                                    else if (type.SpecialType != SpecialType.System_Enum)
```

Comment:
>type.SpecialType != SpecialType.System_Enum [](start = 45, length = 43)

Should we also check for System.ValueType here?

Reply:
System.ValueType is still disallowed as a constraint and will not compile regardless. 
I do not think the errors about invalid mixing are useful/actionable when constraint by itself is not valid.

---
In reply to: [173896814](https://github.com/dotnet/roslyn/pull/25397#discussion_r173896814) [](ancestors = 173896814)